tests/ssg_test_suite/xml_operations.py

```diff
@@ -71,6 +74,71 @@ def infer_benchmark_id_from_component_ref_id(datastream, ref_id):
     return benchmark_node.get('id')
 
 
+@contextlib.contextmanager
+def datastream_root(ds_location, save_location=None):
+    try:
+        tree = ET.parse(ds_location)
+        for prefix, uri in NAMESPACES.items():
+            ET.register_namespace(prefix, uri)
+        root = tree.getroot()
+        yield root
+    finally:
+        if save_location:
+            tree.write(save_location)
+
+
+def remove_machine_platform(root):
+    remove_machine_only_from_element(root, "xccdf:Rule")
+    remove_machine_only_from_element(root, "xccdf:Group")
+
+
+def remove_machine_only_from_element(root, element_spec):
+    query = ".//ds:component/xccdf:Benchmark//{0}".format(element_spec)
+    elements = root.findall(query, NAMESPACES)
+    for el in elements:
+        platforms = el.findall("./xccdf:platform", NAMESPACES)
+        for p in platforms:
+            if p.get("idref") == "cpe:/a:machine":
+                el.remove(p)
+
+
+def get_all_cpe_refs(root):
+    query = ".//ds:component/oval:oval_definitions/oval:definitions/oval:definition[@class='inventory']/oval:metadata/oval:reference[@source='CPE']"
```

Comment:
This is actually querying for the checks used by CPE items, not the CPE items themselves.

Reply:
I am a bit lost, what would be the right thing to do here?

Reply:
Apologies, the code itself is not wrong, but the naming is a bit off. But I may just be nitpicking.

But, I honestly wonder how it is working, because the DS doesn't contain `cpe-item` for other OS versions.
Maybe it is falling back to use the `cpe-dict` from the scanner?
And in this case, all this logic of `add_platform_to_benchmark` wouldn't be necessary...

Reply:
And the OpenSCAP's CPE Dictionary contains rhel7, and other Fedoras:
https://github.com/OpenSCAP/openscap/blob/maint-1.3/cpe/openscap-cpe-dict.xml#L16

Reply:
I need to add something to the benchmark platform. Right now, I get it from the datastream, but what I hear is that I may get the platform CPE string from the `oscap --version` output, i.e. I don't have to search the datastream, hoping to find occurrences of CPE IDs. Do I understand that right?

Reply:
@matejak That is my suspicion.

Because the `platform` entry added doesn't have a counterpart `cpe-item` in the DS. 
So maybe openscap is falling back to its own CPE dictionary, and there it finds the `cpe-item` entry mathing the `platform` added.